extensions/opentelemetry/deployment/src/main/java/io/quarkus/opentelemetry/deployment/OpenTelemetryProcessor.java

```diff
@@ -69,6 +70,14 @@ AdditionalBeanBuildItem ensureProducerIsRetained() {
                 .build();
     }
 
+    @BuildStep
+    void registerNativeImageResources(BuildProducer<ServiceProviderBuildItem> services) throws IOException {
+        services.produce(ServiceProviderBuildItem.allProvidersFromClassPath(
```

Comment:
Shouldn't we also add:
`io.opentelemetry.sdk.autoconfigure.spi.AutoConfigurationCustomizerProvider`
`io.opentelemetry.sdk.autoconfigure.spi.ResourceProvider`
`io.opentelemetry.sdk.autoconfigure.spi.ConfigurablePropagatorProvider`
?

Reply:
I was planning to add official support for them only later. I can add the SPIs, however I'll leave testing and documentation for later because this PR is already too big.

Reply:
Agree. At this point, we should move forward and add these missing pieces later. They are not a blocker at this point.